src/tests/JIT/Regression/JitBlue/Runtime_61629/Runtime_61629.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+public class Runtime_61629
+{
+    public static int Main() => 
+        Test(100, 200.0) + Test(Math.PI, Math.PI) - 72;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static int Test(double a, double b)
```

Comment:
does this fail on current main branch?

Reply:
In Checked builds where we have lots of asserts - yes. For Release one I am not sure

Reply:
I was seeing InvalidProgramException getting thrown for the pattern in the issue in release builds. So likely for this one too.